src/bin/cargo/commands/add.rs

```diff
@@ -188,12 +192,23 @@ pub fn exec(config: &mut Config, args: &ArgMatches) -> CliResult {
 
     let dependencies = parse_dependencies(config, args)?;
 
+    let honor_rust_version = !args.flag("ignore-rust-version");
+    if !config.cli_unstable().msrv_policy && !honor_rust_version {
+        return Err(CliError::new(
+            anyhow::format_err!(
+                "the `ignore-rust-version` flag is unstable, pass -Z msrv-policy to enable it"
```

Comment:
nit: I'd suggest something more like
```rust
"`--ignore-rust-version` is unstable; pass `-Zmsrv-policy` to enable support for it"
```
or look to https://docs.rs/cargo/latest/src/cargo/core/features.rs.html#1034-1057 to see what other unstable errors are like

Reply:
`fail_if_stable_opt` seems more suitable. Should we open a tracking issue separate from the #12043 PR so that it can be referenced in the message?

Reply:
`fail_if_stable_opt` encourages `-Zunstable-opt` rather than letting us use `-Zmsrv-policy`, right?

Reply:
`fail_if_stable_opt` does encourage `-Zunstable-options`, but it does also link to an issue that includes information about the unstable flag.

Side note: if `-Zunstable-options` and `--ignore-rust-version` are present without `-Zmsrv-policy`, the MSRV behavior will be silently skipped. We could emit a warning in this case, but I don't see any precedence for this elsewhere.